lib/shopify_graphql/client.rb

```diff
@@ -113,35 +113,59 @@ def handle_graphql_errors(response)
         else
           ConnectionError.new(response: response)
         end
-      exception.error_code = error_code
```

Comment:
It would be great to keep existing `exception.error_code` as an alias for first error. This will prevent breaking change in the gem and in many cases we only care about single error in the app, so we could continue using `error_code`.